src/Domain/Creators/Shell.hpp

```diff
@@ -90,6 +90,13 @@ class Shell : public DomainCreator<3> {
         "The equatorial compression factor."};
   };
 
+  struct IndexPolarAxis {
```

Comment:
Most of the time we won't apply equatorial compression to the shell domain, and the option could be confusing when it has no effect. I suggest these changes:

- Create a struct `EquatorialCompressionOptions` within the `Shell` class. It can be as simple as the `Object` struct in the BBH domain, just hold two properties `aspect_ratio` and `index_polar_axis`, and be option-creatable, meaning you move the `AspectRatio` and `IndexPolarAxis` options into the struct.
- For the Shell domain, add the option `EquatorialCompression` of type `Options::Auto<EquatorialCompressionOptions, Options::AutoLabel::None>`. Now the user can write:
    ```yaml
    EquatorialCompression: None
    # or
    EquatorialCompression:
      AspectRatio: 2.
      IndexPolarAxis: 0
    ```
- You can choose how to handle the properties within the Shell class. Perhaps it's easiest to just store the `std::optional<EquatorialCompressionOptions>` in the class directly, and retrieve the two parameters when you call `sph_wedges_...`. That's how the BBH domain handles the options for the two objects.